### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by davemcorwin**
@bhennes2 lots of stuff in here, make sure to update your meteor! this includes a reorganization, some updating to new syntax and the introduction of db backed players and games.

### CODE REVIEW

File: .meteor/packages [hunk 1]
```diff
@@ -4,12 +4,28 @@
...
+insecure                # Allow all DB writes from clients (for prototyping)
+
 react
 meteorhacks:npm
 cosmos:browserify
-
-
-npm-container
\ No newline at end of file
+npm-container
```
Comment (bhennes2) [inline .meteor/packages:25]: Do we still need the `react` package?

File: client/components/choose-player-menu.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  render() {
+
+    var menuItems = this.data.players.map((player) => {
+      return { title: player.name };
+    });
+
+    return (
+      <div className="game-container">
+        <div className="player-container left">
+          <p className="player-name">Choose Player</p>
```
Comment (bhennes2) [inline client/components/choose-player-menu.jsx:49]: Should each of these "Choose Player" divs be a React component?

File: client/components/key-handler.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    var myHandleKeyUp = function handleKeyUp(event) {
+      if (event.keyCode !== 65 && event.keyCode !== 66) {
+        return;
+      }
+
+      if (currentKey === event.keyCode) {
+        if (Date.now() - time >= 500) {
+          this.hold(event.keyCode);
+        } else {
+          this.tap(event.keyCode);
```
Comment (bhennes2) [inline client/components/key-handler.jsx:32]: This is to handle "holding" down a key right?

File: client/components/leaders.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    var menuItems = [
+      { title: "Main Menu", target: "/main" }
+    ];
+
+    return (
+      <div>
+        <h1 className="game-name">
+          <img src="/rocket.png" />Leaders
+        </h1>
+        <ul>
```
Comment (bhennes2) [inline client/components/leaders.jsx:20]: Our goal here would be to make this another component eventually, right?

File: client/components/play.jsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  },
+
+  render() {
+
+    var winner = this.state.player1 > this.state.player2 ? this.data.game.player1 : this.data.game.player2;
+    var end = this.state.end ? <EndGameMessage winner={winner}/> : '';
+
+    return (
+      <div className="game-container">
+        <div className="player-container left">
```
Comment (bhennes2) [inline client/components/play.jsx:62]: Do we plan on making these containers a React component as well?

File: client/components/serving-menu.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    var menuItems = [
+      { title: this.data.game.player1, query: {server: 0} },
+      { title: this.data.game.player2, query: {server: 1} }
+    ];
+
+    return (
+      <div className="game-container">
+        <div className="player-container left">
+          <p className="player-name">{this.data.game.player1}</p>
+          <h1 className="score">0</h1>
```
Comment (bhennes2) [inline client/components/serving-menu.jsx:27]: I know you are in progress on a lot of these items :)...just trying to understand the approach.

Will the scores become React components too?

**Comment by bhennes2** [comment]
This is looking good to me. Let's re-group tomorrow for a bit and get this stuff merged in.

File: .meteor/packages (hunk 1 above)
Comment (davemcorwin) [inline .meteor/packages:25]: yes

File: client/components/choose-player-menu.jsx (hunk 2 above)
Comment (davemcorwin) [inline client/components/choose-player-menu.jsx:49]: yes

File: client/components/key-handler.jsx (hunk 3 above)
Comment (davemcorwin) [inline client/components/key-handler.jsx:32]: yes

File: client/components/leaders.jsx (hunk 4 above)
Comment (davemcorwin) [inline client/components/leaders.jsx:20]: yes

**Pull request merged**